x/auth/ante.go

```diff
@@ -149,58 +149,82 @@ func validateBasic(tx StdTx) (err sdk.Error) {
 // if the account doesn't have a pubkey, set it.
 func processSig(
 	ctx sdk.Context, am AccountMapper,
-	addr sdk.AccAddress, sig StdSignature, signBytes []byte) (
+	addr sdk.AccAddress, sig StdSignature, signBytes []byte, simulate bool) (
 	acc Account, res sdk.Result) {
-
 	// Get the account.
 	acc = am.GetAccount(ctx, addr)
 	if acc == nil {
 		return nil, sdk.ErrUnknownAddress(addr.String()).Result()
 	}
 
-	// Check account number.
 	accnum := acc.GetAccountNumber()
-	if accnum != sig.AccountNumber {
-		return nil, sdk.ErrInvalidSequence(
-			fmt.Sprintf("Invalid account number. Got %d, expected %d", sig.AccountNumber, accnum)).Result()
-	}
-
-	// Check and increment sequence number.
 	seq := acc.GetSequence()
-	if seq != sig.Sequence {
-		return nil, sdk.ErrInvalidSequence(
-			fmt.Sprintf("Invalid sequence. Got %d, expected %d", sig.Sequence, seq)).Result()
+
+	// Perform checks that wouldn't pass successfully in simulation, i.e. sig
+	// would be empty as simulated transactions come with no signatures whatsoever.
+	if !simulate {
```

Comment:
Why would we avoid account number and sequence checks in simulation? I think we do want to check those - just not the signature validity.

Reply:
Amended, thanks